sdk/identity/identity/src/credentials/azurePipelinesCredential.ts

```diff
@@ -120,34 +120,40 @@ export class AzurePipelinesCredential implements TokenCredential {
           response.status
         }. Complete response - ${JSON.stringify(response)}`,
       );
-      throw new AuthenticationError(
-        response.status,
-        `${credentialName}: Authenticated Failed. Received null token from OIDC request. Response status- ${
-          response.status
-        }. Complete response - ${JSON.stringify(response)}`,
-      );
+      throw new AuthenticationError(response.status, {
+        error: `${credentialName}: Authenticated Failed. Received null token from OIDC request.`,
+        error_description: `${JSON.stringify(response)}`,
+      });
     }
     try {
       const result = JSON.parse(text);
       if (result?.oidcToken) {
         return result.oidcToken;
       } else {
-        let errorMessage = `${credentialName}: Authentication Failed. oidcToken field not detected in the response.`;
+        const errorMessage = `${credentialName}: Authentication Failed. oidcToken field not detected in the response.`;
```

Comment:
Sorry, one more question. Are you able to figure out where `authority_not_found` is getting set by us and can you help me understand how this PR fixes that scenario?

Reply:
Yes this PR fixes it. The authority_not_found was being set since the error message we were throwing was not in the correct format as required by the `AuthenticationError` constructor